magstock/__init__.py

```diff
@@ -34,9 +34,17 @@ def PREREG_DONATION_OPTS(self):
 
 @Session.model_mixin
 class SessionMixin:
+    def food_consumers(self):
+        """
+        :return: list of anyone who gets food, regardless of if they're comp'd food or if they paid for it
+        """
+        return [a for a in self.query(Attendee).all() if a.gets_food]
+
     def food_purchasers(self):
-        return self.query(Attendee).filter(or_(Attendee.purchased_food == True,
-                                               Attendee.badge_type.in_([c.STAFF_BADGE, c.GUEST_BADGE])))
+        """
+        :return: list of anyone who paid money to get food
+        """
+        return self.query(Attendee).filter(Attendee.purchased_food == True)
```

Comment:
note that the meaning of food_purchasers has now changed.  it used to include anyone getting food (free or not), now includes ONLY people who paid for it.